platforms/core-runtime/base-services/src/main/java/org/gradle/api/JavaVersion.java

```diff
@@ -201,6 +201,18 @@ public static JavaVersion current() {
         return currentJavaVersion;
     }
 
+    /**
+     * Returns the version of latest LTS supported non incubating.
+     *
+     * @return The version of LTS supported non incubating.
+     *
+     * @since 8.7
+     */
+    @Incubating
+    public static JavaVersion getLatestSupportedLTS() {
```

Comment:
We can avoid exposing it to the public API, however, this class is responsible to define the supported java versions which makes more easy to remember to update the latest supported LTS than if we move into any other class like `BuildPropertiesDefaults`

Reply:
In general, we try to avoid adding new public APIs unless necessary. 

If there is no need to add this to the public API right now I would prefer to remove it. 

As this effort continues and we feel there is a strong reason to add this, we can consider it. 

However, for now I do not see strong evidence for adding a public API method for this.

Reply:
Moved directly to BuildPropertiesDefaults.TOOLCHAIN_VERSION